src/chains/ethereum/ethereum/tests/api/debug/debug_storageRangeAt.test.ts

```diff
@@ -134,141 +147,108 @@ describe("api", () => {
-              value:
-                "0x0000000000000000000000000000000000000000000000000000000000000001"
-            }
+          [VALUE3_KEY.hashKey]: {
+            key: VALUE3_KEY.key,
+            value:
+              "0x68656c6c6f20776f726c64000000000000000000000000000000000000000016"
+          },
+          [VALUE2_KEY.hashKey]: {
+            key: VALUE2_KEY.key,
+            value:
+              "0x0000000000000000000000000000000000000000000000000000000000000001"
+          }
         };
         assert.deepStrictEqual(result.storage, storage);
         assert.strictEqual(result.nextKey, null);
       });
 
       it("should return correct storage given a different arbitrary hexadecimal value for startKey", async () => {
-        const result = await provider.send("debug_storageRangeAt", [
-          blockHash,
-          0,
-          contractAddress,
-          "0x290c",
-          3
-        ]);
+        // the start key, 0x290c, sorts _before_ value1.raw, so value1 _is_ still included in the result.
+        const result = await debugStorage(blockHash, 0, 3, "0x290c");
 
         const storage = {
-          "0x290decd9548b62a8d60345a988386fc84ba6bc95484008f6362f93160ef3e563":
-            {
-              key: "0x0000000000000000000000000000000000000000000000000000000000000000",
-              value:
-                "0x0000000000000000000000000000000000000000000000000000000000000005"
-            },
-          "0x405787fa12a823e0f2b7631cc41b3ba8828b3321ca811111fa75cd3aa3bb5ace":
-            {
-              key: "0x0000000000000000000000000000000000000000000000000000000000000002",
-              value:
-                "0x68656c6c6f207270647200000000000000000000000000000000000000000014"
-            },
-          "0xb10e2d527612073b26eecdfd717e6a320cf44b4afac2b0732d9fcbe2b7fa0cf6":
-            {
-              key: "0x0000000000000000000000000000000000000000000000000000000000000001",
-              value:
-                "0x0000000000000000000000000000000000000000000000000000000000000001"
-            }
+          [VALUE1_KEY.hashKey]: {
+            key: VALUE1_KEY.key,
+            value:
+              "0x0000000000000000000000000000000000000000000000000000000000000005"
+          },
+          [VALUE3_KEY.hashKey]: {
+            key: VALUE3_KEY.key,
+            value:
+              "0x68656c6c6f20776f726c64000000000000000000000000000000000000000016"
+          },
+          [VALUE2_KEY.hashKey]: {
+            key: VALUE2_KEY.key,
+            value:
+              "0x0000000000000000000000000000000000000000000000000000000000000001"
+          }
         };
 
         assert.deepStrictEqual(result.storage, storage);
         assert.strictEqual(result.nextKey, null);
       });
 
-      it("should return correct storage given different transaction indexes", async () => {
+      it("should return correct storage given different storage slot changes at different transaction indexes in same block", async () => {
```

Comment:
The updated test!

It now sets a value for the very first time in the middle of a block. This is the exact issue this PR is fixing.